### ISSUE TITLE
the log polling seems to break on the App Dashboard....

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8-console#205]
... works to start with but tends to stop updating

**Comment by gashcrumb** [comment]
Seems odd that this would have broken due to a client-side change, the polling code hasn't changed since May.  Has the data coming back from Jenkins changed I wonder?  From what I see in the javascript console, only the first block of the log is returned, subsequent calls to fetch the log get that same block returned.  Also I see that all the log statements in each new fetch are re-added to the array of lines, so any updates that are showing up is the same initial batch of lines repeated.

**Comment by gashcrumb** [comment]
Interesting, in fact it works fine against our vagrant setup there.  Wonder if the proxying is somehow getting in the way here?

**Comment by jstrachan** [comment]
Ah yeah! Might be worth checking if its just a kubernetes/minikube issue (relating to proxying)?

**Comment by jstrachan** [comment]
FWIW I've not noticed the issue on openshift (minishift) so I think its related to the proxying in kuisp for kubernetes

**Comment by gashcrumb** [comment]
Looked into this more locally, I think the kuisp proxy is winding up returning a cached response.  I even tried switching our request over to a POST which usually gets around that.  This URL should give me the start of the log:

```
http://localhost:9000/k8s/api/v1/proxy/namespaces/default/services/jenkins:http/job/camel-foo1/1/fabric8/logHtml?tail=1&start=0&size=50000
```

which I get, but then the next URL we request should give us the next bit of the log, instead I get back the same response as the previous URL:

```
http://localhost:9000/k8s/api/v1/proxy/namespaces/default/services/jenkins:http/job/camel-foo1/1/fabric8/logHtml?tail=1&start=10240&size=50000&first=0
```

I've added a guard for now in [this commit](https://github.com/hawtio/hawtio-kubernetes/commit/af58f47cfbff363e51339794c8bc41fd3ee3a959) so at least if you happen to leave the page open your browser doesn't blow up :-)

Wonder if we should instead use the Jenkins exposeUrl value to get at the log instead though?  I'll give that a go locally here...

**Comment by jstrachan** [comment]
we'd hit CORS issues if we talk direct though won't we?

**Comment by jstrachan** [comment]
wonder if there's a way to disable caching in kuisp? @jimmidyson - we'd not want caching really as we do that in the client anyways

**Comment by gashcrumb** [comment]
Yup, was just going to post that :-)  Means we'd need to have Jenkins return the right Access-Control-Allow-Origin header.  I think it'd work then though, so there's that :-)

**Comment by jimmidyson** [comment]
Hmm kuisp only adds cache headers for static content. Doesn't cache proxied responses in any way.

**Comment by gashcrumb** [comment]
Just checking here, if I go directly to jenkins for that url I get back different responses based on the parameters, so that bit looks okay...

**Comment by gashcrumb** [comment]
I can't imagine it'd be the API server proxy, unless it's changed recently, this works fine when running against Openshift where we directly connect to the API server.

**Comment by gashcrumb** [comment]
Wonder if we can get jenkins to set 'Cache-Control' to 'no-cache' in it's response.  I tried it from the client but it didn't help.

**Comment by gashcrumb** [comment]
Hey, I noticed the log seems to work today...
